src/Cryptol/ModuleSystem/Base.hs

```diff
@@ -149,8 +150,18 @@ parseModule path = do
               }
 
   case P.parseModule cfg txt of
-    Right pm -> let fp = fingerprint bytes
-                in fp `seq` return (fp, pm)
+    Right pm ->
+      do let fp = fingerprint bytes
+         pm1 <- case path of
+                  InFile p ->
+                    do r <- getByteReader
+                       mb <- io (removeIncludesModule r p pm)
+                       case mb of
+                         Right ok -> pure ok
+                         Left err -> noIncludeErrors err
+                  _ -> pure pm
```

Comment:
If you migrate the comment, it would be better to match on `InMem` explicitly here to make the connection more obvious.